src/address-book/AddressBook.test.tsx

```diff
@@ -0,0 +1,74 @@
+import '@testing-library/jest-dom/extend-expect'
+import React from 'react'
+import Web3 from 'web3'
+import {fireEvent, render, waitFor} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import {AddressBook} from './AddressBook'
+import {WalletState, WalletStatus} from '../common/wallet-state'
+import {SettingsState} from '../settings-state'
+import {BackendState} from '../common/backend-state'
+import {expectNoValidationErrorOnSubmit} from '../common/test-helpers'
+
+const web3 = new Web3()
+
+it('adds a new contact', async () => {
+  const initialState = {walletStatus: 'LOADED' as WalletStatus, web3}
+  const {queryByText, getByLabelText, getByTestId} = render(
+    <SettingsState.Provider>
+      <WalletState.Provider initialState={initialState}>
+        <BackendState.Provider initialState={{web3}}>
+          <AddressBook />
+        </BackendState.Provider>
+      </WalletState.Provider>
+    </SettingsState.Provider>,
+  )
+
+  const newContactButton = getByTestId('add-contact-button')
+  userEvent.click(newContactButton)
+
+  // dialog opened
+  await waitFor(() => expect(queryByText('Edit Contact')).toBeInTheDocument())
+
+  const saveButton = getByTestId('right-button')
+  expect(saveButton).toBeDisabled()
+
+  const cancelButton = getByTestId('left-button')
+  expect(cancelButton).toBeEnabled()
+
+  // address validation works
+  const addressInput = getByLabelText('Address')
+  const INVALID_ADDRESS_MSG = 'Invalid address'
+
+  fireEvent.change(addressInput, {target: {value: 'not an address'}})
+  await waitFor(() => expect(queryByText(INVALID_ADDRESS_MSG)).toBeInTheDocument())
+
+  const FINAL_ADDRESS = '0xffffffffffffffffffffffffffffffffffffffff'
+  fireEvent.change(addressInput, {target: {value: FINAL_ADDRESS}})
+  await waitFor(() => expect(queryByText(INVALID_ADDRESS_MSG)).not.toBeInTheDocument())
+
+  expect(saveButton).toBeDisabled()
+
+  // label validation works
+  const labelInput = getByLabelText('Label')
+  const INVALID_LABEL_MSG = 'Label length must not exceed 80 characters'
+  const LIMIT = 80
```

Comment:
Can't we just export `LABEL_MAX_LIMIT` from `AddressBook.tsx`?

Reply:
It wouldn't protect from an accidental change in `LABEL_MAX_LIMIT`.
For the same reason it's preferable to hardcode the strings in tests, otherwise an accidental change in translations could slip in. It doesn't really make sense to test for "import x, assert x == x".